teps/0096-pipelines-v1-release.md

```diff
@@ -0,0 +1,104 @@
+
+## Motivation
+
+To become the industry standard, cloud-native CI/CD platform,
+Tekton must provide users with a stable API they can rely on.
+Some users may be waiting for V1 stability before deciding to use Tekton.
+
+### Goals
+
+- Define stability policy for Pipelines V1.
+- Define supported and unsupported use cases for Pipelines V1.
+- Define criteria for a feature to be "required" or "optional" for Pipelines V1.
+- Define what updates, if any, are needed to the [Pipelines API definition](https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md) before V1 release.
+
+### Non Goals
+
+- Define the stability or V1 plans of other Tekton project.
+- Discuss aspects of Pipelines unrelated to the API, such as performance, dogfooding, release automation,
+and testing and debugging features.
+
+## Background
+
+Tekton currently uses Kubernetes' [API versioning guidelines](https://kubernetes.io/docs/reference/using-api/api-overview/#api-versioning)
+to define API stability levels, as described in the [compatibility policy](https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md#alpha-beta-and-ga).
+
+Tekton API components may contain features at a lower level of stability, gated behind feature flags,
+as described in [TEP-0033: Tekton Feature Gates](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-feature-gates.md).
+
+Both Tekton and Kubernetes software releases use [semantic versioning](https://semver.org), but Tekton does not follow 
+Kubernetes' guidelines for [release versioning](https://github.com/kubernetes/sig-release/blob/master/release-engineering/versioning.md).
+Specifically, Tekton does not include release tags containing "alpha", "beta", or "rc".
+
+Tekton also follows Kubernetes' [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/). 
+However, while Kubernetes defines stability periods in terms of months or releases, whichever is longer,
+Tekton defines stability periods only in terms of months. 
+
+| API Version | Kubernetes Deprecation Policy | Tekton Deprecation Policy |
+|:----------- |:----------------------------- |:------------------------- |
+| Alpha       | 0 releases                    | 0 months                  |
+| Beta        | 9 months or 3 releases        | 9 months 
+
+## Proposal
+
+### V1 Stability and Deprecation Policy
+
+Tekton will continue to follow the Kubernetes [API versioning guidelines](https://kubernetes.io/docs/reference/using-api/api-overview/#api-versioning) and [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/). Users will have at least 12 months to upgrade to backwards incompatible changes, meaning that a previous API version must be supported for 12 months from when a new API version is created. Backwards incompatible API changes must be accompanied by a major version software release, deprecation warnings, and migration instructions from the previous version. 
```

Comment:
When we say "Backwards incompatible API changes must be accompanied by a major version software release", do we means v2.x.y for example ? With APIs version, we should be able to provide a backward incompatible API (`v2alpha1` for example) independently of the major version in the release, right ? I would tend to agree though, that if we were to publish a `v2` API version, it should most likely require a major version.

Reply:
Added an example. In light of TEP 33, if we want to add an experimental feature, for example, we would add it behind an alpha feature flag, and it could be removed without changing the API version or software major version number. However, if we changed or removed a stable feature, that would require updating the API version from V1 to V2.

Reply:
moved to non goals!